### PR TITLE
Implement MapEntities for ActionDiffEvent

### PR DESCRIPTION
**Pull request by zhaop**
Implementing `MapEntities` for `ActionDiffEvent`s allows networking crates like replicon to automatically translate the ID of the "owner entity" for action diffs from the client's set of entity IDs to the server's (or vice versa).

------------------

I'm new to contributing to LWIM, should I open an issue first? Let me know if anything else is needed :)

### CODE REVIEW

File: src/action_diff.rs [hunk 1]
```diff
@@ -62,6 +62,16 @@ pub struct ActionDiffEvent<A: Actionlike> {
     pub action_diffs: Vec<ActionDiff<A>>,
 }
 
+/// Implements entity mapping for `ActionDiffEvent`.
+///
+/// This allows the owner entity to be remapped when transferring event diffs
+/// between different ECS contexts (e.g. client and server).
```
Comment (cBournhonesque) [inline src/action_diff.rs:68]:
```suggestion
/// between different ECS worlds (e.g. client and server).
```

**Review by cBournhonesque — APPROVED** [review]
LGTM

**Comment by zhaop** [comment]
Any preferences regarding merging main into PR vs rebasing PR onto main?

**Comment by alice-i-cecile** [comment]
No preferences on the strategy there :) Merging main in is generally less error-prone IME.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
I'll merge this once CI is passing <3

**Pull request merged**